pkg/plugins/runtime/k8s/metadata/annotations.go

```diff
@@ -22,6 +22,11 @@ const (
 	// inbound listeners won't be generated in that case
 	KumaGatewayAnnotation = "kuma.io/gateway"
 
+	// KumaBuiltinGatewayAnnotation marks a pod as Kuma Gateway dataplane
+	// element. This annotation should only ever be applied by a Kuma control
+	// plane. It alters the dataplane generated by the pod controller.
+	KumaBuiltinGatewayAnnotation = "gateways.kuma.io/builtin-name"
```

Comment:
IIUC, this annotation is the name of the gateway that a dataplane will provide capacity for. I think we should call this `gateways.kuma.io/gateway-name` in that case.